src/Symfony/Component/ObjectMapper/Metadata/ReverseClassObjectMapperMetadataFactory.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\Metadata;
+
+use Symfony\Component\ObjectMapper\Attribute\Map;
+
+/**
+ * @author Florent Blaison <florent.blaison@gmail.com>
+ */
+final class ReverseClassObjectMapperMetadataFactory implements ObjectMapperMetadataFactoryInterface
+{
+    /**
+     * @var array<string, Mapping[]>
+     */
+    private array $attributesCache = [];
+
+    public function __construct(
+        private readonly ObjectMapperMetadataFactoryInterface $objectMapperMetadataFactory,
+        /** @param array<int, class-string> $classMap */
```

Comment:
```suggestion
        /** @param array<class-string, class-string> $classMap */
```
Makes more sense to me.
Also, can you put it on top of the constructor?